include/sys/onoff.h

```diff
@@ -456,77 +335,47 @@ static inline void onoff_client_init_callback(struct onoff_client *cli,
  * passes a pointer object associated with an incomplete service
  * operation.
  *
- * @retval Non-negative on successful (initiation of) request
- * @retval -EIO if service has recorded an an error
- * @retval -EINVAL if the parameters are invalid
- * @retval -EAGAIN if the reference count would overflow
+ * @retval non-negative the observed state of the machine at the time
+ * the request was processed, if successful.
+ * @retval -EIO if service has recorded an an error.
+ * @retval -EINVAL if the parameters are invalid.
+ * @retval -EAGAIN if the reference count would overflow.
  * @retval -EWOULDBLOCK if the function was invoked from non-thread
  * context and successful initiation could result in an attempt to
  * make the calling thread sleep.
  */
-int onoff_request(struct onoff_service *srv,
+int onoff_request(struct onoff_manager *mgr,
 		  struct onoff_client *cli);
 
 /**
  * @brief Release a reserved use of an on-off service.
  *
- * The return value indicates the success or failure of an attempt to
- * initiate an operation to release the resource.  If initiation of
- * the operation succeeds the result of the release operation itself
- * is provided through the configured client notification method,
- * possibly before this call returns.
- *
- * Note that the call to this function may succeed in a case where the
- * actual release fails.  Always check the operation completion
- * result.
- *
- * @param srv the service that will be used.
- *
- * @param cli a non-null pointer to client state providing
- * instructions on how to notify the client when release completes.
- * Behavior is undefined if cli references an object associated with
- * an incomplete service operation.
- *
- * @retval Non-negative on successful (initiation of) release
- * @retval -EINVAL if the parameters are invalid
- * @retval -EIO if service has recorded an an error
- * @retval -EWOULDBLOCK if a non-blocking request was made and
- *         could not be satisfied without potentially blocking.
- * @retval -EALREADY if the service is already off or transitioning
- *         to off
- * @retval -EBUSY if the service is transitioning to on
- */
-int onoff_release(struct onoff_service *srv,
-		  struct onoff_client *cli);
-
-/**
- * @brief Test whether an on-off service has recorded an error.
+ * @note Behavior is undefined if this is not paired with a preceding
+ * onoff_request() call that completed successfully.
  *
- * This function can be used to determine whether the service has
- * recorded an error.  Errors may be cleared by invoking
- * onoff_service_reset().
+ * @param mgr the manager for which a request was successful.
  *
- * @return true if and only if the service has an uncleared error.
+ * @retval non-negative the observed state (ONOFF_STATE_ON) of the
+ * machine at the time of the release, if the release succeeds.
+ * @retval -EIO if service has recorded an an error.
+ * @retval -ENOTSUP if the machine is not in a state that permits release.
```

Comment:
imo `-EALREADY` would be better. Similar to cancelation: `-EALREADY if cli was not a record of an uncompleted
 notification`

Reply:
I think that would be misleading.  It's not that this specific client has already released (where `-EALREADY` would be appropriate), it's that the machine is in a state where no client can be released: off, transitioning to off, transitioning to on, resetting.  In all those cases invoking this is a true error, not a redundant operation.

Reply:
All other comments have been addressed.